battery

```diff
@@ -75,26 +75,18 @@ battery_charge() {
             fi
             ;;
         "Linux")
-            case $(cat /etc/*-release) in
-                *"Arch Linux"*|*"Ubuntu"*|*"openSUSE"*)
-                    battery_state=$(cat $battery_path/energy_now)
-                    battery_full=$battery_path/energy_full
-                    battery_current=$battery_path/energy_now
-                    ;;
-                *)
-                    battery_state=$(cat $battery_path/status)
-                    battery_full=$battery_path/charge_full
-                    battery_current=$battery_path/charge_now
-                    ;;
-            esac
-            if [ $battery_state == 'Discharging' ]; then
-                BATT_CONNECTED=0
-            else
-                BATT_CONNECTED=1
-            fi
-                now=$(cat $battery_current)
-                full=$(cat $battery_full)
-                BATT_PCT=$((100 * $now / $full))
+	         for battery_path in ${battery_paths[@]}; do
+                         battery_state=$(cat $battery_path/status)
+			 if [[ $battery_state == "Charging" || $battery_full == "Full" ]]; then
+		 	     BATT_CONNECTED=1
+			 fi
+                         battery_full=$battery_path/energy_full
+                         battery_current=$battery_path/energy_now
```

Comment:
Hello, quick question: Why did you remove the `Arch Linux` , `Ubuntu` and `openSUSE` checks? are these not necessary at all?

Reply:
Hi, i wasn’t sure about the differences in the first place, but on my test machines (Arch and Ubuntu) both paths were the same as the second case.